lib/content/utils.js

```diff
@@ -1,74 +1,126 @@
-   *
-   * @returns {boolean} false-not in uuid format / true-it is!
-   */
-  id: (req) => {
-    if ((typeof req.params.id !== 'string') || (!isUUID(req.params.id))) {
-      _.set(req.session, '404', {
-        message: config.content.messages.format.id,
-        safe: `/${config.content.base}`,
-      });
+  if (!typ || typ === '') {
+    return config.content.messages.missing.type.replace('%type', '(no type given)');
+  }
 
-      return false;
-    }
+  const exists = allTypes.find(tp => {
+    return tp.id === typ.toLowerCase();
+  });
 
-    return true;
-  },
+  if (!exists || (exists === false) || (typeof exists !== 'object') || !exists.id || !exists.name) {
+    return config.content.messages.missing.type.replace('%type', req.params.type.toString());
+  }
 
-  /**
-   * Determine if a revision is a number; update session for error
-   *
-   * @param {object} req - HTTP Request
-   *
-   * @returns {boolean} false-not number / true-it is!
-   */
-  revision: (req) => {
-    if (!Number.isInteger(parseInt(req.params.revision, 10))) {
-      _.set(req.session, '404', {
-        message: config.content.messages.format.revision.replace('%revision', req.params.revision),
-        safe: `/${config.content.base}`,
-      });
-
-      return false;
-    }
+  return exists;
+};
 
-    return true;
-  },
+/**
+ * Determine if an id is in uuid format
+ *
+ * @param {string} ident - content type id
+ *
+ * @returns {boolean|string} true-id is string and in uuid format / string = error message
+ */
+const id = (ident) => {
+  if (ident === '' || typeof ident !== 'string' || !isUUID(ident)) {
+    return config.content.messages.format.id;
+  }
+
+  return true;
+};
+
+/**
+ * Determine if a revision is a number
+ *
+ * @param {number} rev - revision number
+ *
+ * @returns {boolean} false-not number / true-it is!
+ */
+const revision = (rev) => {
+  if (rev === '' || !Number.isInteger(parseInt(rev, 10))) {
+    return config.content.messages.format.revision;
+  }
+
+  return true;
+};
 
-  /**
-   * If type is false, update session for error
-   *
-   * @param {object} req - HTTP Request
-   * @param {object} type - full content type object
-   *
-   * @returns {boolean} false-not true / true-it is!
-   */
-  type: (req, type) => {
-    if (!type || (type === false) || (typeof type !== 'object') || !type.id || !type.name) {
-      _.set(req.session, '404', {
-        message: config.content.messages.missing.type.replace('%type', req.params.type),
-        safe: `/${config.content.base}`,
-      });
-
-      return false;
+/*
+ * @typedef routeDetails
+ * @type object
+ *
+ * @property {object} type - Content type configuration, after merged with input plugins
+ * @property {object} workflow - workflow configuration
+ * @property {string} id - id of a piece of content
+ * @property {string} revision - revision number for a revision of a piece of content
+ */
+
+/**
+ * Checking that what is in a content url matches what's in the system
```

Comment:
This description needs to be changed. My assumption was that it was going to return true/false, not that it was going to return an object gleaned from the URL.

Also,  not sure if I like that it's returning a string with an error as opposed to causing an error of some sort. This feels like a good use of a promise as an interface for us, where we either resolve with the object or reject with an error string instead of a sync function that a user will need to have checks in place themselves to determine if it's passed or failed